### ISSUE TITLE
RangeError: Error while loading rule 'prefer-const': Maximum call stack size exceeded

### ISSUE THREAD

**Issue by catch6** [antfu/eslint-flat-config-utils#5]
### Describe the bug
ESLint 'prefer-const' rule causes maximum call stack size exceeded
 
eslint.config.mjs
```js
import antfu from '@antfu/eslint-config'

export default antfu({
  formatters: true,
  vue: true,
}, {
  rules: {
    'import/consistent-type-specifier-style': 'off',
    'no-console': 'off',
    'vue/html-self-closing': 'off',
    'antfu/top-level-function': 'off',
  },
})
```
package version
```
    node: v22.16.0
    "@antfu/eslint-config": "^5.0.0",
    "eslint": "^9.32.0",
    "eslint-plugin-format": "^1.0.1",
```

error logs
```
[Info  - 14:06:03] ESLint server is starting.
[Info  - 14:06:03] ESLint server running in node v20.19.0
[Info  - 14:06:03] ESLint server is running.
[Info  - 14:06:20] ESLint library loaded from: /Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint@9.32.0_jiti@2.4.2/node_modules/eslint/lib/api.js
[@antfu/eslint-config] Detected running in editor, some rules are disabled.
[Error - 10:10:18] An unexpected error occurred:
[Error - 10:10:18] RangeError: Error while loading rule 'prefer-const': Maximum call stack size exceeded
Occurred while linting /Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/app/components/course/Play.vue
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
[Error - 10:10:18] An unexpected error occurred:
[Error - 10:10:18] RangeError: Error while loading rule 'prefer-const': Maximum call stack size exceeded
Occurred while linting /Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/app/components/course/Play.vue
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
[Error - 10:10:21] An unexpected error occurred:
[Error - 10:10:21] RangeError: Error while loading rule 'prefer-const': Maximum call stack size exceeded
Occurred while linting /Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/app/components/course/Play.vue
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:20:10)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
[Error - 10:10:34] An unexpected error occurred:
[Error - 10:10:34] RangeError: Error while loading rule 'prefer-const': Maximum call stack size exceeded
Occurred while linting /Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/app/components/course/Play.vue
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:20:10)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
[Error - 10:14:20] An unexpected error occurred:
[Error - 10:14:20] RangeError: Error while loading rule 'prefer-const': Maximum call stack size exceeded
Occurred while linting /Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/app/components/course/Play.vue
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:20:10)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
    at Object.get (file:///Users/zh/Projects/VSCodeProjects/mindit/zhiyu-web/node_modules/.pnpm/eslint-flat-config-utils@2.1.0/node_modules/eslint-flat-config-utils/dist/index.mjs:32:24)
    at Reflect.get (<anonymous>)
```

### Reproduction

Occasionally, this situation occurs, and it returns to normal after restarting VSCode, making it difficult to fully reproduce.

### System Info

```Shell
System:
    OS: macOS 15.5
    CPU: (10) arm64 Apple M1 Pro
    Memory: 164.09 MB / 16.00 GB
    Shell: 5.9 - /bin/zsh
  Binaries:
    Node: 22.16.0 - ~/.local/state/fnm_multishells/80822_1754115322695/bin/node
    npm: 10.9.2 - ~/.local/state/fnm_multishells/80822_1754115322695/bin/npm
    pnpm: 10.13.1 - ~/.pnpm/pnpm
    bun: 1.2.9 - ~/.bun/bin/bun
  Browsers:
    Chrome: 138.0.7204.184
    Safari: 18.5
```

### Used Package Manager

npm

### Validations

- [x] Follow our [Code of Conduct](https://github.com/antfu/.github/blob/main/CODE_OF_CONDUCT.md)
- [x] Read the [Contributing Guide](https://github.com/antfu/contribute).
- [x] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [x] Check that this is a concrete bug. For Q&A, please open a GitHub Discussion instead.
- [x] The provided reproduction is a [minimal reproducible](https://stackoverflow.com/help/minimal-reproducible-example) of the bug.

### Contributions

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests (actually just go ahead and do it, thanks!)

**Comment by romansp** [comment]
This is a duplicate of #4 and has been fixed in https://github.com/antfu/eslint-flat-config-utils/commit/1ff6d063237428234220e4913152c0d94f9c325c.

@antfu sorry to tag you but can you please update `eslint-flat-config-utils` dependency in `@antfu/eslint-config`? Thanks!
